src/utils/allocators.zig

```diff
@@ -1,123 +1,140 @@
 const std = @import("std");
+const builtin = @import("builtin");
 const sig = @import("../sig.zig");
 
-pub const RecycleFBA = struct {
-    // this allocates the underlying memory + dynamic expansions
-    // (only used on init/deinit + arraylist expansion)
-    backing_allocator: std.mem.Allocator,
-    // this does the data allocations (data is returned from alloc)
-    alloc_allocator: std.heap.FixedBufferAllocator,
-    // recycling depot
-    records: std.ArrayList(Record),
-
-    // for thread safety
-    // TODO: add a config option to enable threadsafe in all the methods
-    // instead of having to manually lock the allocator outside of the method calls
-    mux: std.Thread.Mutex = .{},
-
-    const Record = struct { is_free: bool, buf: [*]u8, len: u64 };
-    const Self = @This();
-
-    pub fn init(backing_allocator: std.mem.Allocator, n_bytes: u64) !Self {
-        const buf = try backing_allocator.alloc(u8, n_bytes);
-        const alloc_allocator = std.heap.FixedBufferAllocator.init(buf);
-        const records = std.ArrayList(Record).init(backing_allocator);
+pub fn RecycleFBA(config: struct {
```

Comment:
More of a question: Any reason to have this as a type function? given that it does not take any "generic" parameter

Reply:
> given that it does not take any "generic" parameter

type function does not need to take "generic" parameter. They can take parameter of `type` or any compile time known value.